transformations/chinese_butter_fingers_perturbation/transformation.py

```diff
@@ -0,0 +1,227 @@
+import itertools
+import random
+import unidecode as u
+import json
+import os
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+import shutil
+import jieba
+import pypinyin
+
+
+def chinese_butter_finger(text,
+                          chinese_character_database,
+                          common_chinese_character_database,
+                          chinese_words_database,
+                          prob,
+                          rare_word_prob,
+                          seed,
+                          max_outputs,
+                          consider_tone,
+                          word_level_perturb):
+    random.seed(seed)
+
+    perturbed_texts = []
+    for _ in itertools.repeat(None, max_outputs):
+        butter_text = ""
+        output = jieba.lcut(text)
+        if(word_level_perturb):
+            words_and_similar_word_dict_list = get_words_with_similar_pinyin(output,
+                                                                            rare_word_prob,
+                                                                            chinese_character_database,
+                                                                            common_chinese_character_database,
+                                                                            chinese_words_database,
+                                                                            consider_tone)
+            for dict in words_and_similar_word_dict_list:
+                perturb_word = dict['perturb_word']
+                similar_pinyins_words = dict['similar_pinyin_words']
+                if random.random() <= prob and len(similar_pinyins_words) != 0:
+
+                    new_chinese_character = random.choice(similar_pinyins_words)
+                else:
+                    new_chinese_character = perturb_word
+                butter_text += new_chinese_character
+        else:
+            for chinese_character in text:
+                similar_pinyins = get_characters_with_similar_pinyin(chinese_character,
```

Comment:
I didn't use a list for the char-level transformation since a string with all the characters with the same pinyin is sufficient for me to work with. I can convert them to lists though if you think that would be better?

Reply:
it was to be coherent with the previous function but that is okay as such.